[dcp_poc] Introduce a new simple rank local checkpointer

teja-rao opened the pull request:
Summary:
Adds an experimental implementation for a rank local checkpointer with save and load with partial load, blind load and in-place load.

This uses an new API and simpler format.

Plan to add async checkpointing, IO layer, pluggable storage backend, layout customization,  Resharding, deduplication etc are not implemented.

Test Plan: unit tests

Reviewed By: saumishr

Differential Revision: D75426560


cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k

saumishr reviewed (approved):
Great work!! @teja-rao

teja-rao committed:
[dcp_poc] Introduce a new simple rank local checkpointer (#156142)

Summary:

Adds an experimental implementation for a rank local checkpointer with save and load with partial load, blind load and in-place load. 

This uses an new API and simpler format. 

Plan to add async checkpointing, IO layer, pluggable storage backend, layout customization,  Resharding, deduplication etc are not implemented.

Test Plan: unit tests

Reviewed By: saumishr

Differential Revision: D75426560
